src/index.js

```diff
@@ -23,103 +41,220 @@ export const get = (obj, ...paths) => {
 export const themeGet = (path, fallback = null) => props =>
   get(props.theme, path, fallback)
 
-export const is = n => n !== undefined && n !== null
-
-export const isObject = n => typeof n === 'object' && n !== null
+function merge(acc, item) {
+  if (!item) {
+    return acc
+  }
 
-export const num = n => typeof n === 'number' && !isNaN(n)
+  return deepmerge(acc, item, {
+    clone: false, // No need to clone deep, it's way faster.
+  })
+}
 
-export const px = n => (num(n) && n !== 0 ? n + 'px' : n)
+// Style
+function callOrReturn(fn, arg) {
+  if (!func(fn)) return fn
+  const next = fn(arg)
+  return callOrReturn(next, arg)
+}
 
-export const createMediaQuery = n => `@media screen and (min-width: ${px(n)})`
+function getThemeValue(theme, path, initial) {
+  if (!theme) return undefined
+  return callOrReturn(get(initial || theme, path, undefined), { theme })
```

Comment:
Is this to support new functionality?

Reply:
Yeah, it is a new feature (you can release it hidden for now), it permits to have reference values in theme.

```js
const theme = {
  red: 'red',
  primary: ({ theme }) => theme.red,
}
```